extension.ts

```diff
@@ -12,37 +12,79 @@ import { ModeHandler } from './src/mode/modeHandler';
 import { TaskQueue } from './src/taskQueue';
 import { Position } from './src/motion/position';
 
+export class EditorIdentity {
+  private _fileName: string;
+  private _viewColumn: vscode.ViewColumn;
+
+  constructor(textEditor?: vscode.TextEditor) {
+    this._fileName = textEditor && textEditor.document.fileName || "";
+    this._viewColumn = textEditor && textEditor.viewColumn || vscode.ViewColumn.One;
+  }
+
+  get fileName() {
+    return this._fileName;
+  }
+
+  get viewColumn() {
+    return this._viewColumn;
+  }
+
+  public hasSameBuffer(identity: EditorIdentity): boolean {
+    return this.fileName === identity.fileName;
+  }
+
+  public isEqual(identity: EditorIdentity): boolean {
+    return this.fileName === identity.fileName && this.viewColumn === identity.viewColumn;
+  }
+
+  public withColumn(column: vscode.ViewColumn): EditorIdentity {
```

Comment:
Oh it's not used for now, I thought this can be used to update Identity by column but finally it's not yet necessary. If other parts are good, I'll remove it and push a update.